src/Control/Effect/Cut.hs

```diff
@@ -15,15 +15,13 @@ module Control.Effect.Cut
 , run
 ) where
 
-import Control.Applicative (Alternative(..))
 import Control.Effect.Carrier
 import Control.Effect.NonDet
 import Control.Monad (MonadPlus(..))
-import Control.Monad.Fail
+import qualified Control.Monad.Fail as Fail
 import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
-import Prelude hiding (fail)
```

Comment:
The above also means that we don’t have to hide the import of `fail` from `Prelude` now.